src/nevergreen/wrap_content_security_policy.clj

```diff
@@ -1,21 +1,23 @@
 (ns nevergreen.wrap-content-security-policy
   (:require [clojure.string :refer [starts-with? join]]
+            [nevergreen.config :as config]
             [ring.util.response :refer [get-header]]))
 
-(def ^:private default-src "default-src 'self'")
-(def ^:private script-src "script-src 'self' 'unsafe-eval'")
-(def ^:private worker-src "worker-src 'self'")
-(def ^:private style-src "style-src 'self' 'unsafe-inline'")
-(def ^:private img-src "img-src * data:")
-(def ^:private font-src "font-src 'self' data:")
-(def ^:private media-src "media-src *")
-(def ^:private connect-src "connect-src 'self' https://api.github.com https://gist.githubusercontent.com")
-(def ^:private object-src "object-src 'none'")
-(def ^:private child-src "child-src 'self'")
-(def ^:private frame-src "frame-src 'none'")
-(def ^:private sources [default-src script-src worker-src style-src img-src font-src media-src connect-src object-src child-src frame-src])
+(defn- csp-headers []
+  ["default-src 'self'"
+   "script-src 'self' 'unsafe-eval'"
+   "worker-src 'self'"
+   "style-src 'self' 'unsafe-inline'"
+   "img-src * data:"
+   "font-src 'self' data:"
+   "media-src *"
+   "connect-src 'self' https://api.github.com https://gist.githubusercontent.com"
+   "object-src 'none'"
+   (str "frame-ancestors " (config/allow-iframe-from))])
 
 (defn wrap-content-security-policy [app]
   (fn [req]
     (let [res (app req)]
-      (assoc-in res [:headers "Content-Security-Policy"] (join "; " sources)))))
+      (-> res
+          (assoc-in [:headers "X-Frame-Options"] nil)
```

Comment:
I'm dropping the `X-Frame-Options` entirely as it takes the whitelisted URLs in a different scheme format than the CSP one. This does leave older browsers vulnerable to clickjacking.

We could change this to optionally remove the header based off the config.

Reply:
We officially only support the last 2 versions of browsers anyway (for adding CSS prefixes and polyfills when compiling the UI)